runner/src/artifacts.rs

```diff
@@ -47,6 +48,8 @@ pub enum Artifact {
     Downloaded { name: String, url: String },
     /// Artifact available as binaries on the local file system.
     Binary { path: PathBuf },
+    /// Disk artifact, used to run various linux distributions
+    Disk { name: String, url: String },
```

Comment:
So `Artifact` was intended for `Binary` initially, I don't think the disk abstraction fits well here.

Ideally I would recommend renaming `Artifact` to `BinaryArtifact` and creating a new, separate `DiskArtifact`. That way we can handle them separately, and in particular treat the payload and the disk differently.